recipes/dgl/meta.yaml

```diff
@@ -0,0 +1,87 @@
+{% set name = "dgl" %}
+{% set version = "1.1.0" %}
+{% set torch_proc_type = "cuda" if cuda_compiler_version != "None" else "cpu" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/dmlc/dgl
+  git_rev: {{ version }}
+  patches:
+    - conda-build.patch
+
+build:
+  skip: true  # [not linux]
+  number: 0
+  rpaths:
+    - lib/
+      # PyTorch libs are in site-packages intead of with other shared objects
+    - {{ SP_DIR }}/torch/lib/
+  missing_dso_whitelist:
+    - '*/libcuda.*'  # [linux64]
+  string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}  # [cuda_compiler_version not in (undefined, 'None')]
+    - sysroot_{{ target_platform }} 2.17  # [linux]
+    - cmake
+    - make
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
+    - git 
+
+
+  host:
+    - python =3.10
+    - setuptools
+    - cython <3
+    - dlpack ==0.7
+    - dmlc ==0.5
+    - gklib ==5.1.1
```

Comment:
Appears this can't be removed otherwise the build fails. Please see this comment for details: https://github.com/conda-forge/staged-recipes/pull/22691#issuecomment-1652408390

Reply:
Any statically linked libraries need to have their license included in the license section of this recipe.

There are some possibilities I can think of:

1. DGL is linking to the static library object produced by the METIS submodule instead of the GKlib conda package (but you shouldn't need this submodule because you are using the conda package?)
2. DGL is linking to a header-only portion of GKlib?
3. DGL is missing a GKlib argument during the linking step of the build, so conda cannot detect any links to from DGL to GKlib, but at runtime an undefined symbol error will occur
4. DGL doesn't link to GKlib. It only links to METIS, but METIS mentions GKlib in its headers, so the headers have to be there at build time but not runtime.

Reply:
What I will do is switch to making the sub modules explicit as recommended here https://github.com/conda-forge/staged-recipes/pull/22691#discussion_r1275477390

Then be sure to package the license files as asked to here: https://github.com/conda-forge/staged-recipes/pull/22691#discussion_r1275479206